pageserver/src/tenant.rs

```diff
@@ -3215,6 +3215,18 @@ impl Tenant {
             }
         }
 
+        if let ShutdownMode::Reload = shutdown_mode {
+            tracing::info!("Flushing deletion queue");
+            if let Err(e) = self.deletion_queue_client.flush().await {
+                if !matches!(e, DeletionQueueError::ShuttingDown) {
```

Comment:
Since ShuttingDown is the only variant of this enum, one can avoid writing a log that will never be output by doing something like
`match e {ShuttingDown => {// we don't care}}`, that way it just becomes a compile time error if we ever add more variants